tests/e2e/vsphere_shared_datastore.go

```diff
@@ -121,4 +126,158 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] "+
 		gomega.Expect(errorOccurred).To(gomega.BeTrue())
 
 	})
+
+	// Verify impact on existing pv pvc when sc recreated with different binding mode
+	// Steps
+	// 1. Create a Storage Class with Immediate Binding Mode.
+	// 2. Create a PVC using above SC.
+	// 3. Wait for PVC to be in Bound phase.
+	// 4. Create standalone pod
+	// 5. Delete Storgae Class created above
+	// 6. Recreate Storage Class with same name as above but with Binding mode WaitForFirstConsumer
+	// 7. Check PVC status after recreating storage class
+	// 8. Check pod status after recreating storage class
+	// 9. Delete Pod
+	// 10. Delete PVC and SC
+
+	ginkgo.It("[csi-block-vanilla] [csi-guest] [csi-supervisor] "+
+		"Verify impact on existing pv pvc when sc recreated with different binding mode", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Invoking Test to verify impact on existing pv pvc when sc recreated with different binding mode")
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		if !vanillaCluster {
+			storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores2)
+		}
+		// Create Storage class and PVC
+		ginkgo.By("Creating Storage Class and PVC")
+		if guestCluster {
+			scParameters[svStorageClassName] = storagePolicyName
+			storageclass, pvclaim, err = createPVCAndStorageClass(client,
+				namespace, nil, scParameters, diskSize, nil, "", false, "")
+		} else if supervisorCluster {
+			namespace = getNamespaceToRunTests(f)
+			profileID := e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+			scParameters[scParamStoragePolicyID] = profileID
+			createResourceQuota(client, namespace, rqLimit, storagePolicyName)
+			storageclass, pvclaim, err = createPVCAndStorageClass(client,
+				namespace, nil, scParameters, diskSize, nil, "", false, "", storagePolicyName)
+		} else if vanillaCluster {
+			scParameters[scParamFsType] = ext4FSType
+			storageclass, pvclaim, err = createPVCAndStorageClass(client,
+				namespace, nil, scParameters, "", nil, "", false, "")
+		}
+		framework.Logf("storageclass name :%s", storageclass.GetName())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			if supervisorCluster {
```

Comment:
1. You are first deleting the storage class and then updating to immediate . Is this working?
2. In both SVC and GC update the storage class to immediate - instead of doing it here add this in defer block after line 240 . i.e, after updating the SC to WFC

Reply:
Reply to Point1 - You are first deleting the storage class and then updating to immediate . Is this working?

As suggested, I am using storage class "shared-ds-policy-2" (Initially it will be in "Immediate Binding Mode")
Later I am updating "shared-ds-policy-2" to "WaitForFirstConsumer" Binding mode.

If any older testcases is using this "shared-ds-policy-2", hence, I am deleting it and creating it again with "Immediate Binding Mode". This scenario I have tested and its working. (Attached are the test case results).

I am not sure if "shared-ds-policy-2" is  used by any other older testcase, If not, I will delete this SC "shared-ds-policy-2" as a part of cleanup process.

Reply:
Point 2 - In both SVC and GC update the storage class to immediate - instead of doing it here add this in defer block after line 240 . i.e, after updating the SC to WFC

acknowledged. Will make this change.